### PR TITLE
restore: fix the bug that gc life time ttl does not take effect

### LINKED ISSUE #449: Lightning failed  in checksum phase due to incorrect gc lifetime
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
Use lightning v4.0.8 to load a single table about 2~3T SQL files. Lightning failed at the checksum phase with following error:
```
2020-11-03T11:15:33.486581055+00:00 tinewswriter lightning [2020/11/03 11:15:32.625 +00:00] [ERROR] [checksum.go:145] ["remote checksum failed"] [table=banking_transactions] [takeTime=11m16.884037533s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2020-11-03 11:04:15 +0000 UTC, GC safe point is 2020-11-03 11:05:09.401 +0000 UTC"]
2020-11-03T11:15:33.486637705+00:00 tinewswriter lightning [2020/11/03 11:15:32.626 +00:00] [ERROR] [restore.go:693] ["restore table failed"] [table=`newswriter`.`banking_transactions`] [takeTime=18h28m4.073806043s] [error="restore table `newswriter`.`banking_transactions` failed: [tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2020-11-03 11:04:15 +0000 UTC, GC safe point is 2020-11-03 11:05:09.401 +0000 UTC"]
2020-11-03T11:15:33.486694558+00:00 tinewswriter lightning [2020/11/03 11:15:32.629 +00:00] [ERROR] [restore.go:810] ["restore all tables data failed"] [takeTime=18h41m33.914314719s] [error="restore table `newswriter`.`banking_transactions` failed: [tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2020-11-03 11:04:15 +0000 UTC, GC safe point is 2020-11-03 11:05:09.401 +0000 UTC"]
```

2. What did you expect to see?

3. What did you see instead?

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):

        ```
        (paste TiDB-Lightning version here)
        ```
        
    - TiKV-Importer version (run `tikv-importer -V`)

        ```
        (paste TiKV-Importer version here)
        ```
        
    - TiKV version (run `tikv-server -V`):

        ```
        (paste TiKV version here)
        ```

    - TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `inventory.ini` if deployed by Ansible

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for TiDB-Lightning if possible

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Fix the gc lifetime doesn't take effect because the backend routine to periodically update gc ttl is not started.

close #449 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

**Review by kennytm — COMMENTED** [review]
LGTM

**Comment by overvenus** [comment]
Please update your PR title. https://github.com/pingcap/community/blob/01457b0bba90b92a6d8f6c715a3b18cc432c6a12/contributors/commit-message-pr-style.md

File: lightning/restore/checksum.go [hunk 1]
```diff
@@ -331,6 +338,7 @@ func (m *gcTTLManager) addOneJob(ctx context.Context, table string, ts uint64) e
 	if curTs == 0 || m.currentTs < curTs {
 		return m.doUpdateGCTTL(ctx, m.currentTs)
 	}
+
```
Comment (overvenus) [inline lightning/restore/checksum.go:341]:
```suggestion
```

**Review by overvenus — COMMENTED** [review]
Please add a unit test

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**